Sources/Shared/Requests/Request+PublishToSocial.swift

```diff
@@ -0,0 +1,106 @@
+    static func fetchPublishJob(for videoID: String) -> Request {
+        let publishToSocialPath = String.videosURI + "/\(videoID)" + String.publishToSocialURI
+        return Request(path: publishToSocialPath)
+    }
+
+    /// Returns a `put` request for publishing `SocialMediaPosts` for the given video ID.
+    /// - Note: Post data should be validated prior to creating the publish request. Posts that fail server-side
+    ///         validation will produce an error for that platform.
+    /// - Parameters:
+    ///   - posts: A structure containing all of the data necessary to publish to multiple platforms simultaneously.
+    ///   - videoID: The identifier for the video to be published.
+    static func publishPosts(_ posts: SocialMediaPosts, for videoID: String) -> Request {
+        let publishToSocialPath = String.videosURI + "/\(videoID)" + String.publishToSocialURI
+        var parameters = [String: Any]()
+
+        posts.facebook.map {
+            var post = [String: Any]()
+            post[String.Key.title] = $0.title
+            post[String.Key.description] = $0.description
+            post[String.Key.pageID] = $0.pageID
+            post[String.Key.categoryID] = $0.categoryID
+            post[String.Key.allowEmbedding] = $0.allowEmbedding
+            post[String.Key.shouldAppearOnNewsFeed] = $0.shouldAppearOnNewsFeed
+            post[String.Key.isSecretVideo] = $0.isSecretVideo
+            post[String.Key.allowSocialActions] = $0.allowSocialActions
+            parameters[String.Key.facebook] = post
+        }
+
+        posts.linkedIn.map {
+            var post = [String: Any]()
+            post[String.Key.pageID] = $0.pageID
+            post[String.Key.title] = $0.title
+            post[String.Key.description] = $0.description
+            parameters[String.Key.linkedin] = post
+        }
+
+        posts.twitter.map {
+            var post = [String: String]()
+            post[String.Key.tweet] = $0.tweet
+            parameters[String.Key.twitter] = post
+        }
+
+        posts.youTube.map {
+            var post = [String: Any]()
+            post[String.Key.title] = $0.title
+            post[String.Key.description] = $0.description
+            post[String.Key.privacy] = $0.privacy.rawValue
+            post[String.Key.categoryID] = $0.categoryID
+            $0.tags.map { (tags) in  post[String.Key.tags] = tags }
+            parameters[String.Key.youtube] = post
+        }
+
+        return Request(method: .put, path: publishToSocialPath, parameters: parameters)
+    }
+}
+
+private extension String {
+    static let videosURI = "/videos"
+    static let publishToSocialURI = "/publish_to_social"
+
+    struct Key {
+        static let facebook = "facebook"
+        static let linkedin = "linkedin"
+        static let twitter = "twitter"
+        static let youtube = "youtube"
+        static let title = "title"
+        static let description = "description"
+        static let pageID = "page_id"
```

Comment:
`page_id` used for linkedin. For facebook need add `destiantion` key
https://github.vimeows.com/Vimeo-API/design/pull/148/files#diff-1f1bcce0884b841f7fd76d22ef138caeR27